samcli/commands/pipeline/init/interactive_init_flow.py

```diff
@@ -0,0 +1,244 @@
+import shutil
+from pathlib import Path
+from typing import Dict, List
+
+import click
+
+from samcli.cli.main import global_cfg
+from samcli.commands.exceptions import PipelineTemplateCloneException
+from samcli.lib.cookiecutter.interactive_flow import InteractiveFlow
+from samcli.lib.cookiecutter.interactive_flow_creator import InteractiveFlowCreator
+from samcli.lib.cookiecutter.template import Template
+from samcli.lib.utils.git_repo import GitRepo, CloneRepoException
+from samcli.lib.utils.osutils import rmtree_callback
+from .pipeline_templates_manifest import PipelineTemplateManifest, PipelineTemplatesManifest
+
+LOG = logging.getLogger(__name__)
+shared_path: Path = global_cfg.config_dir
+APP_PIPELINE_TEMPLATES_REPO_URL = "https://github.com/aws/aws-sam-cli-pipeline-init-templates.git"
+APP_PIPELINE_TEMPLATES_REPO_LOCAL_NAME = "aws-sam-cli-app-pipeline-templates"
+CUSTOM_PIPELINE_TEMPLATE_REPO_LOCAL_NAME = "custom-pipeline-template"
+
+
+def do_interactive() -> None:
+    """
+    An interactive flow that prompts the user for pipeline template (cookiecutter template) location, downloads it,
+    runs its specific questionnaire then generates the pipeline config file based on the template and user's responses
+    """
+    click.echo("Which pipeline template source would you like to use?")
+    click.echo("\t1 - AWS Quick Start Pipeline Templates\n\t2 - Custom Pipeline Template Location")
+    location_choice = click.prompt("Choice", type=click.Choice(["1", "2"]), show_choices=False)
+    if location_choice == "2":
+        _generate_from_custom_location()
+    else:
+        _generate_from_app_pipeline_templates()
+
+
+def _generate_from_app_pipeline_templates() -> None:
+    """
+    Prompts the user to choose a pipeline template from SAM predefined set of pipeline templates hosted in the git
+    repository: aws/aws-sam-cli-pipeline-init-templates.git
+    downloads locally, then generates the pipeline config file from the selected pipeline template.
+    """
+    pipeline_templates_local_dir: Path = _clone_app_pipeline_templates()
+
+    pipeline_templates_manifest: PipelineTemplatesManifest = _read_app_pipeline_templates_manifest(
+        pipeline_templates_dir=pipeline_templates_local_dir
+    )
+    # The manifest contains multiple pipeline-templates so select one
+    selected_pipeline_template_manifest: PipelineTemplateManifest = _select_pipeline_template(
+        pipeline_templates_manifest=pipeline_templates_manifest
+    )
+    selected_pipeline_template_dir: Path = pipeline_templates_local_dir.joinpath(
+        selected_pipeline_template_manifest.location
+    )
+    _generate_from_pipeline_template(pipeline_template_dir=selected_pipeline_template_dir)
+
+
+def _generate_from_custom_location() -> None:
+    """
+    Prompts the user for a custom pipeline template location, downloads locally, then generates the pipeline config file
+    """
+    pipeline_template_repo_url: str = click.prompt(text="template Git location")
```

Comment:
How does this show up in the terminal? Should it be:

```suggestion
    pipeline_template_repo_url: str = click.prompt(text="Template Git location: ")
```

Reply:
the colon is automatically added by click